bluebutton-data-pipeline-fhir-load/src/main/java/gov/hhs/cms/bluebutton/datapipeline/fhir/transform/DataTransformer.java

```diff
@@ -929,6 +948,85 @@ private TransformedBundle transformPartDEvent(RifRecordEvent<PartDEventRow> rifR
 	}
 
 	/**
+	 * @param rifRecordEvent
+	 *            the source {@link RifRecordEvent} to be transformed
+	 * @return the {@link TransformedBundle} that is the result of transforming
+	 *         the specified {@link RifRecordEvent}
+	 */
+	private TransformedBundle transformCarrierClaim(RifRecordEvent<CarrierClaimGroup> rifRecordEvent) {
+		if (rifRecordEvent == null)
+			throw new IllegalArgumentException();
+		CarrierClaimGroup claimGroup = rifRecordEvent.getRecord();
+		if (1 != claimGroup.version)
+			throw new IllegalArgumentException("Unsupported record version: " + claimGroup.version);
+		if (claimGroup.recordAction != RecordAction.INSERT)
+			// Will need refactoring to support other ops.
+			throw new BadCodeMonkeyException();
+
+		Bundle bundle = new Bundle();
+
+		ExplanationOfBenefit eob = new ExplanationOfBenefit();
+		eob.setId("ExplanationOfBenefit/" + claimGroup.claimId);
+		eob.getCoverage().setCoverage(referenceCoverage(claimGroup.beneficiaryId, COVERAGE_PLAN_PART_B));
+		eob.setPatient(referencePatient(claimGroup.beneficiaryId));
+
+		setPeriodStart(eob.getBillablePeriod(), claimGroup.dateFrom);
+		setPeriodEnd(eob.getBillablePeriod(), claimGroup.dateThrough);
+
+		/*
+		 * Referrals are represented as contained resources, because otherwise
+		 * updating them would require an extra roundtrip to the server (can't
+		 * think of an intelligent client-specified ID for them).
+		 */
+		Practitioner referrer = new Practitioner();
+		referrer.addIdentifier().setSystem(CODING_SYSTEM_NPI_US).setValue(claimGroup.referringPhysicianNpi);
+		Reference referrerRef = referencePractitioner(claimGroup.referringPhysicianNpi);
+		upsert(bundle, referrer, referrerRef.getReference());
+		ReferralRequest referral = new ReferralRequest();
```

Comment:
This makes sense to me, thanks for the explanation! Does this work for any resource then that we would expect to have the same lifetime as its referring EOB resource? Would it make sense for a MedicationOrder, for instance, to be linked to the EOB in such a way?